contracts/express/ExpressExecutable.sol

```diff
@@ -3,112 +3,249 @@
+                )
+            ) revert NotApprovedByGateway();
+        }
+        address expressCaller = _popExpressCallerWithToken(
+            commandId,
+            sourceChain,
+            sourceAddress,
+            payload,
+            tokenSymbol,
+            amount
+        );
+        if (expressCaller != address(0)) {
+            {
+                address gatewayToken = gateway.tokenAddresses(tokenSymbol);
+                IERC20(gatewayToken).safeTransfer(expressCaller, amount);
+            }
+            emit ExpressExecutionWithTokenFulfilled(
+                commandId,
+                sourceChain,
+                sourceAddress,
+                payload,
+                tokenSymbol,
+                amount,
+                expressCaller
+            );
+        } else {
+            _executeWithToken(sourceChain, sourceAddress, payload, tokenSymbol, amount);
+        }
     }
 
-    /**
-     * @dev Defines the logic to execute the call.
-     * @param sourceChain The source blockchain.
-     * @param sourceAddress The source address.
-     * @param payload The payload.
-     * This is a virtual function and needs to be implemented in the derived contracts.
-     */
-    function _execute(
+    function expressExecute(
+        bytes32 commandId,
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload
+    ) external payable {
+        if (gateway.isCommandExecuted(commandId)) revert AlreadyExecuted();
+
+        address expressCaller = msg.sender;
+
+        _setExpressCaller(commandId, sourceChain, sourceAddress, payload, expressCaller);
+        _execute(sourceChain, sourceAddress, payload);
+        emit ExpressExecuted(commandId, sourceChain, sourceAddress, payload, expressCaller);
+    }
+
+    function expressExecuteWithToken(
+        bytes32 commandId,
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload,
+        string calldata symbol,
+        uint256 amount
+    ) external payable {
+        if (gateway.isCommandExecuted(commandId)) revert AlreadyExecuted();
+        address expressCaller = msg.sender;
+        {
+            address gatewayToken = gateway.tokenAddresses(symbol);
+            IERC20(gatewayToken).safeTransferFrom(expressCaller, address(this), amount);
+        }
+        _setExpressCallerWithToken(commandId, sourceChain, sourceAddress, payload, symbol, amount, expressCaller);
+        _executeWithToken(sourceChain, sourceAddress, payload, symbol, amount);
+        emit ExpressExecutedWithToken(commandId, sourceChain, sourceAddress, payload, symbol, amount, expressCaller);
+    }
+
+    function _expressExecutionSlot(
+        bytes32 commandId,
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload
+    ) internal pure returns (uint256 slot) {
+        // TODO: maybe add some salt although i doubt it will be an issue.
```

Comment:
yes, should add a prefix since we don't know how the application will do storage